src/jrd/Database.cpp

```diff
@@ -616,7 +649,8 @@ namespace Jrd
 		m_eventMgr = nullptr;
```

Comment:
On enter `g_mutex` should be locked. 
At line 639 the `shutdownMutex` is locked.
So we have order of mutexes: `g_mutex` then `shutdownMutex`.

At line 643 `g_mutex` is unlocked, while `shutdownMutex` still locked.
At line 646 `g_mutex` will be locked again and we have inverted order of mutexes: `shutdownMutex` then `g_mutex`.
This is a way for deadlock.

Am I wrong ?

Reply:
I added comments mentioning the possibility of a deadlock situation arising in the future. However, if you believe the probability is too high, I can look into the different solution once more.